app/controllers/user_subscriptions_controller.rb

```diff
@@ -0,0 +1,87 @@
+class UserSubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def subscribed
+    params.require(%i[source_type source_id])
+    source_type = params[:source_type]
+    source_id = params[:source_id]
+
+    is_subscribed = UserSubscriptions::CacheChecker.new(current_user, source_type, source_id).cached_subscription_check
+
+    render json: {
+      is_subscribed: is_subscribed,
+      status: 200
```

Comment:
Isn't this a bit redundant with the HTTP status? If we want to communicate success, how about a boolean flag `success: true` which potentially also simplifies the JS consumer a bit.

Reply:
That's fair, it is. I'll switch it to a boolean.

My best guess for why we do this in other places is because once you resolve the promise in JS and parse the response body, the function you pass that parsed response to no longer is aware of what the status was. By having the status code duplicated, it avoided having to pass it separately. I don't plan on differentiating logic by specific codes so boolean seems good to me 👍.